### PR TITLE
[pytorch][build][ec2] PyTorch 1.12.1 EC2 release

### PR DESCRIPTION
**Pull request by ShiboXing**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
EC2 images release of Pytorch 1.12.1, bug fix on the` test_eks_pytorch_dgl_single_node_training`

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [x] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/training/docker/1.12/py3/cu113/Dockerfile.gpu [hunk 1]
```diff
@@ -4,11 +4,11 @@ ARG PYTHON_SHORT_VERSION=3.8
 ARG MAMBA_VERSION=4.12.0-2
```
Comment (ydaiming) [inline pytorch/training/docker/1.12/py3/cu113/Dockerfile.gpu:4]: Why do you need to change this cu113 for E3? E3 only uses cu116 for PT 1.12

File: pytorch/buildspec.yml [hunk 2]
```diff
@@ -46,21 +46,37 @@ context:
...
+    device_type: &DEVICE_TYPE cpu
+    python_version: &DOCKER_PYTHON_VERSION py3
+    tag_python_version: &TAG_PYTHON_VERSION py38
+    os_version: &OS_VERSION ubuntu20.04
+    tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *OS_VERSION, "-ec2" ]
+    docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /Dockerfile., *DEVICE_TYPE ]
+    target: ec2
+    context:
+      <<: *TRAINING_CONTEXT  
+  BuildEC2GPUPTTrainPy3DockerImageCU113:
```
Comment (ydaiming) [inline pytorch/buildspec.yml:63]: What is this for?

File: pytorch/buildspec.yml [hunk 3]
```diff
@@ -46,20 +46,20 @@ context:
...
+    image_size_baseline: 5000
+    device_type: &DEVICE_TYPE cpu
+    python_version: &DOCKER_PYTHON_VERSION py3
+    tag_python_version: &TAG_PYTHON_VERSION py38
+    os_version: &OS_VERSION ubuntu20.04
+    tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *OS_VERSION, "-ec2" ]
+    docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /Dockerfile., *DEVICE_TYPE ]
+    target: ec2
+    context:
+      <<: *TRAINING_CONTEXT
```
Comment (ydaiming) [inline pytorch/buildspec.yml:62]: Nit: remove space

File: pytorch/inference/docker/1.12/py3/cu116/Dockerfile.gpu [hunk 4]
```diff
@@ -2,10 +2,10 @@ ARG PYTHON=python3
 ARG PYTHON_VERSION=3.8.13
 
 # PyTorch Binaries
-ARG PT_INFERENCE_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.12.0_e3/20220628-223427/c1e155d3cbd8d4115c80866bed4c717775363e7a/torch-1.12.0%2Bcu116-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHVISION_URL=https://download.pytorch.org/whl/cu116/torchvision-0.13.0%2Bcu116-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHAUDIO_URL=https://download.pytorch.org/whl/cu116/torchaudio-0.12.0%2Bcu116-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHDATA_URL=https://download.pytorch.org/whl/test/torchdata-0.4.0-cp38-cp38-linux_x86_64.whl
+ARG PT_INFERENCE_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.12.1_e3/20220826-044615/579ffd0893a9d3e60773918bef0b000dfe60a7ce/torch-1.12.1%2Bcu116-cp38-cp38-linux_x86_64.whl
+ARG PT_TORCHVISION_URL=https://download.pytorch.org/whl/cu116/torchvision-0.13.1%2Bcu116-cp38-cp38-linux_x86_64.whl
```
Comment (ydaiming) [inline pytorch/inference/docker/1.12/py3/cu116/Dockerfile.gpu:6]: Nit: remove space

File: pytorch/training/docker/1.12/py3/cu116/Dockerfile.gpu [hunk 5]
```diff
@@ -4,11 +4,11 @@ ARG PYTHON_SHORT_VERSION=3.8
...
-ARG PT_EC2_TRAINING_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.12.0_e3/20220628-223427/c1e155d3cbd8d4115c80866bed4c717775363e7a/torch-1.12.0%2Bcu116-cp38-cp38-linux_x86_64.whl
-ARG PT_SM_TRAINING_URL=https://aws-pytorch-cicd-v3-binaries.s3.us-west-2.amazonaws.com/r1.11.0_v3_sm/20220404-193901/15c50b561b1a638dacc59db932cab0fcb8d60d10/gpu/torch-1.11.0%2Bcu113-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHVISION_URL=https://download.pytorch.org/whl/cu116/torchvision-0.13.0%2Bcu116-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHAUDIO_URL=https://download.pytorch.org/whl/cu116/torchaudio-0.12.0%2Bcu116-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHDATA_URL=https://download.pytorch.org/whl/test/torchdata-0.4.0-cp38-cp38-linux_x86_64.whl
+ARG PT_EC2_TRAINING_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.12.1_e3/20220826-044615/579ffd0893a9d3e60773918bef0b000dfe60a7ce/torch-1.12.1%2Bcu116-cp38-cp38-linux_x86_64.whl
+ARG PT_SM_TRAINING_URL=https://download.pytorch.org/whl/cu116/torch-1.12.1%2Bcu116-cp38-cp38-linux_x86_64.whl
+ARG PT_TORCHVISION_URL=https://download.pytorch.org/whl/cu116/torchvision-0.13.1%2Bcu116-cp38-cp38-linux_x86_64.whl
```
Comment (ydaiming) [inline pytorch/training/docker/1.12/py3/cu116/Dockerfile.gpu:9]: Nit: remove space

File: pytorch/training/docker/1.12/py3/cu113/Dockerfile.gpu (hunk 1 above)
Comment (ydaiming) [inline pytorch/training/docker/1.12/py3/cu113/Dockerfile.gpu:4]: Revert changes in this file.

**Review by ydaiming — DISMISSED** [review]
Looks good if build and tests pass

File: pytorch/inference/docker/1.12/py3/cu116/Dockerfile.gpu [hunk 6]
```diff
@@ -2,10 +2,10 @@ ARG PYTHON=python3
 ARG PYTHON_VERSION=3.8.13
 
 # PyTorch Binaries
-ARG PT_INFERENCE_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.12.0_e3/20220628-223427/c1e155d3cbd8d4115c80866bed4c717775363e7a/torch-1.12.0%2Bcu116-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHVISION_URL=https://download.pytorch.org/whl/cu116/torchvision-0.13.0%2Bcu116-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHAUDIO_URL=https://download.pytorch.org/whl/cu116/torchaudio-0.12.0%2Bcu116-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHDATA_URL=https://download.pytorch.org/whl/test/torchdata-0.4.0-cp38-cp38-linux_x86_64.whl
+ARG PT_INFERENCE_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.12.1_e3/20220826-044615/579ffd0893a9d3e60773918bef0b000dfe60a7ce/torch-1.12.1%2Bcu116-cp38-cp38-linux_x86_64.whl
```
Comment (arjkesh) [inline pytorch/inference/docker/1.12/py3/cu116/Dockerfile.gpu:5]: We are no longer using the "e3" name and are now using "ec2" instead. Can you please change the name of these artifacts to reflect that change?

File: pytorch/training/docker/1.12/py3/Dockerfile.cpu [hunk 7]
```diff
@@ -3,11 +3,11 @@ ARG PYTHON_VERSION=3.8.13
 ARG MAMBA_VERSION=4.12.0-2
 
 # PyTorch Binaries
-ARG PT_EC2_TRAINING_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.12.0_e3/20220714-051536/64a46e47aea7de3d99f75c5b4feba6d66c80af8b/torch-1.12.0%2Bcpu-cp38-cp38-linux_x86_64.whl
-ARG PT_SM_TRAINING_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.12.0_sm/20220713-224008/9f169f548f4b1eca6631c07d48bbe73a59e300ca/torch-1.12.0%2Bcpu-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHVISION_URL=https://download.pytorch.org/whl/cpu/torchvision-0.13.0%2Bcpu-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHAUDIO_URL=https://download.pytorch.org/whl/cpu/torchaudio-0.12.0%2Bcpu-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHDATA_URL=https://download.pytorch.org/whl/test/torchdata-0.4.0-cp38-cp38-linux_x86_64.whl
+ARG PT_EC2_TRAINING_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.12.1_e3/20220826-041141/579ffd0893a9d3e60773918bef0b000dfe60a7ce/torch-1.12.1%2Bcpu-cp38-cp38-linux_x86_64.whl
```
Comment (arjkesh) [inline pytorch/training/docker/1.12/py3/Dockerfile.cpu:6]: We are no longer using the "e3" name and are now using "ec2" instead. Can you please change the name of these artifacts to reflect that change?

File: pytorch/training/docker/1.12/py3/cu116/Dockerfile.gpu [hunk 8]
```diff
@@ -4,11 +4,11 @@ ARG PYTHON_SHORT_VERSION=3.8
 ARG MAMBA_VERSION=4.12.0-2
 
 # PyTorch Binaries
-ARG PT_EC2_TRAINING_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.12.0_e3/20220628-223427/c1e155d3cbd8d4115c80866bed4c717775363e7a/torch-1.12.0%2Bcu116-cp38-cp38-linux_x86_64.whl
-ARG PT_SM_TRAINING_URL=https://aws-pytorch-cicd-v3-binaries.s3.us-west-2.amazonaws.com/r1.11.0_v3_sm/20220404-193901/15c50b561b1a638dacc59db932cab0fcb8d60d10/gpu/torch-1.11.0%2Bcu113-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHVISION_URL=https://download.pytorch.org/whl/cu116/torchvision-0.13.0%2Bcu116-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHAUDIO_URL=https://download.pytorch.org/whl/cu116/torchaudio-0.12.0%2Bcu116-cp38-cp38-linux_x86_64.whl
-ARG PT_TORCHDATA_URL=https://download.pytorch.org/whl/test/torchdata-0.4.0-cp38-cp38-linux_x86_64.whl
+ARG PT_EC2_TRAINING_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.12.1_e3/20220826-044615/579ffd0893a9d3e60773918bef0b000dfe60a7ce/torch-1.12.1%2Bcu116-cp38-cp38-linux_x86_64.whl
```
Comment (arjkesh) [inline pytorch/training/docker/1.12/py3/cu116/Dockerfile.gpu:7]: We are no longer using the "e3" name and are now using "ec2" instead. Can you please change the name of these artifacts to reflect that change?

File: pytorch/buildspec.yml [hunk 9]
```diff
@@ -90,68 +90,68 @@ images:
...
+  BuildEC2CPUPTInferencePy3DockerImage:
+    <<: *INFERENCE_REPOSITORY
+    build: &PYTORCH_CPU_INFERENCE_PY3 false
+    # enable_test_promotion: false
+    image_size_baseline: 4899
+    device_type: &DEVICE_TYPE cpu
+    python_version: &DOCKER_PYTHON_VERSION py3
+    tag_python_version: &TAG_PYTHON_VERSION py38
+    os_version: &OS_VERSION ubuntu20.04
+    tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *OS_VERSION, "-e3" ]
```
Comment (arjkesh) [inline pytorch/buildspec.yml:134]: this should be "ec2" and not "e3" - please ensure that no other occurrences of "ec2" were changed back to "e3"

**Review by arjkesh — COMMENTED** [review]
Please update the PR to change "e3" to "ec2" - including in the description. Additionally, please fill out the description of the PR

**Review by tejaschumbalkar — APPROVED** [review]

**Review by jeet4320 — APPROVED** [review]

**Pull request merged**